### PR TITLE
Fix most HLint warnings

### PR DESCRIPTION
**Pull request by ianbollinger**
This leaves in place 10 warnings, mostly for reducing duplication.

### CODE REVIEW

File: src/Language/PureScript/Publish/ErrorsWarnings.hs [hunk 1]
```diff
@@ -182,10 +182,10 @@ displayUserError e = case e of
       , spacer
       ] ++ spdxExamples ++
       [ spacer
-      , para (concat
-          [ "Note that distributing code without a license means that nobody "
-          , "will (legally) be able to use it."
-          ])
+      , para (
```
Comment (hdgarrood) [inline src/Language/PureScript/Publish/ErrorsWarnings.hs]: Does HLint not warn for the other instances of `concat` + a list literal? Changing this makes it inconsistent within the file. I don't really mind which approach we take but I do want it to be consistent within this file.

Comment (ianbollinger) [inline src/Language/PureScript/Publish/ErrorsWarnings.hs]: It only complains if it's given a list literal with two elements.

**Comment by paf31** [comment]
@hdgarrood Good to merge then?

**Comment by hdgarrood** [comment]
I'd still rather keep it consistent within that file that I commented on. Other than that, it looks great.

**Comment by ianbollinger** [comment]
Er, it was already inconsistent within that file. (Using either ++ or concat.) But I can revert the edits if you like.

**Comment by hdgarrood** [comment]
How is it inconsistent currently?

**Comment by ianbollinger** [comment]
[Here](https://github.com/purescript/purescript/blob/master/src/Language/PureScript/Publish/ErrorsWarnings.hs#L101), [here](https://github.com/purescript/purescript/blob/master/src/Language/PureScript/Publish/ErrorsWarnings.hs#L126), [here](https://github.com/purescript/purescript/blob/master/src/Language/PureScript/Publish/ErrorsWarnings.hs#L232), and [here](https://github.com/purescript/purescript/blob/master/src/Language/PureScript/Publish/ErrorsWarnings.hs#L268).

**Comment by hdgarrood** [comment]
Ah, wasn't aware of those. Might as well merge as-is then.

**Comment by paf31** [comment]
Thanks!

**Pull request merged**